lib/perl/Genome/Carp.pm

```diff
@@ -0,0 +1,41 @@
+package Genome::Carp;
+
+use strict;
+use warnings;
+
+use Carp qw();
+use Exporter qw(import);
+use Sub::Install qw(install_sub);
+
+my @carp_methods = qw(carpf croakf confessf cluckf longmessf shortmessf);
+our @EXPORT_OK = (@carp_methods, 'dief', 'warnf');
+
+for my $method_name (@carp_methods) {
+    my $carp_method_name = substr($method_name, 0, -1);
+    my $carp_method = Carp->can($carp_method_name);
+    unless ($carp_method) {
+        die 'method not found: Carp::' . $carp_method_name;
+    }
+
+    install_sub({
+        code => sub {
+            my ($template, @args) = @_;
+            local $Carp::CarpLevel = 1;
```

Comment:
I think you always have to increment it.  In this test, Dier::do_die sets its own CarpLevel to make the message report its caller's caller (GrandparentCaller::calls_do_die), but the reported location is Caller::calls_do_die, because ```croakf``` is stomping over the existing CarpLevel.  If you swap the two ```local``` lines in croakf, then you get the location the caller wants.

```
use Carp qw();

package Dier;

sub do_die {
    local $Carp::CarpLevel = 1;
    Genome::Carp::croakf('in do_die');
}

package Caller;

sub calls_do_die {
    Dier::do_die();
}

package GrandparentCaller;

sub calls_do_die {
    Caller::calls_do_die();
}

package Genome::Carp;

sub croakf {
    my($tmpl, @params) = @_;
    local $Carp::CarpLevel = 1;  # Existing, wrong
#    local $Carp::CarpLevel =  $Carp::CarpLevel + 1;  # Right
    Carp::croak(sprintf($tmpl, @params));
}

package main;

GrandparentCaller::calls_do_die();
```